rocketpy/mathutils/function.py

```diff
@@ -2612,6 +2622,77 @@ def compose(self, func, extrapolate=False):
                 extrapolation=self.__extrapolation__,
             )
 
+    def _check_user_input(
+        self,
+        source,
+        inputs,
+        outputs,
+        interpolation,
+        extrapolation,
+    ):
+        # check source for data type
+
+        # if list or ndarray, check for dimensions, interpolation and extrapolation
+        if isinstance(source, (list, np.ndarray)):
+            # this will also trigger an error if the source is not a list of numbers or if the array is not homogeneous
+            source = np.array(source, dtype=np.float64)
+
+            # check dimensions
+            source_dim = source.shape[1]
+
+            # check output type and dimensions
+            if isinstance(outputs, str):
+                outputs = [outputs]
+
+            elif len(outputs) > 1:
+                raise ValueError(
+                    f"Output must either be a string or have dimension 1, it currently has dimension ({len(outputs)})."
+                )
+
+            # check interpolation and extrapolation
+            if source_dim > 2:
+                # check for inputs and outputs
+                if inputs == ["Scalar"]:
+                    inputs = [f"Input {i+1}" for i in range(source_dim - 1)]
+                    warnings.warn(
+                        f"Inputs not set, defaulting to {inputs} for multidimensional functions.",
+                    )
+
+                if interpolation != None and interpolation != "shepard":
+                    interpolation = "shepard"
+                    warnings.warn(
+                        "Interpolation method for multidimensional functions is set to 'shepard', currently other methods are not supported.",
```

Comment:
Small comment but still valuable...
Some comment and warning message lines are larger than 88 characters. We'd appreciate if you break it into more lines please.

Reply:
Updated